server/src/main/java/org/apache/druid/server/coordinator/helper/DruidCoordinatorBalancer.java

```diff
@@ -106,8 +108,18 @@ private void balanceTier(
       return;
     }
 
-    final List<ServerHolder> toMoveFrom = Lists.newArrayList(servers);
-    final List<ServerHolder> toMoveTo = Lists.newArrayList(servers);
+    // Sort order is ascending based on available size. We reverse the "from" to shed the largest nodes first
+    final List<ServerHolder> toMoveFrom;
+    final List<ServerHolder> toMoveTo;
+
+    final int nodeLimit = params.getCoordinatorDynamicConfig().getBalancerNodeLimit();
+    if (nodeLimit > 0) {
+      toMoveFrom = Lists.reverse(randomSortedSubList(servers, nodeLimit));
```

Comment:
I wonder why we need to sort and reverse, as I understand the balancing strategy picks segments to move randomly from the server pool?

Reply:
You are correct, I miss-read how some of the sampling stuff works.